com.chartboost.mediation/Runtime/Plugins/iOS/ChartboostMediationUnityBridge.mm

```diff
@@ -616,7 +616,27 @@ void _chartboostMediationSetUserIdentifier(const char * userIdentifier)
 }
 
 float _chartboostMediationGetUIScaleFactor() {
-    // TODO: https://github.com/ChartBoost/ios-helium-sdk/pull/1314
+    // `UIScreen.main` was deprecated in iOS 16. Apple doc:
+    //   https://developer.apple.com/documentation/uikit/uiscreen/1617815-main
+    // Since `UIScreen.main` has been working correctly at least up to iOS 16, the custom
+    // implementation only targets iOS 17+, not iOS 13+.
+    if (@available(iOS 17.0, *)) {
+        NSSet<UIScene*> *connectedScenes = UIApplication.sharedApplication.connectedScenes;
+        NSArray *activationStates = @[@(UISceneActivationStateForegroundActive), @(UISceneActivationStateForegroundInactive), @(UISceneActivationStateBackground), @(UISceneActivationStateUnattached)];
+        for (NSNumber *activationState in activationStates) {
+            UISceneActivationState state = (UISceneActivationState)activationState.integerValue;
+            for (UIScene* connectedScene in connectedScenes) {
+                UIWindowScene *windowScene = (UIWindowScene*)connectedScene;
+                if (windowScene) {
+                    if (windowScene.activationState == state) {
+                        return windowScene.screen.scale;
```

Comment:
This is quite the navigation to get the screen scale. iOS really doesn't have public methods to get these of a root scene? 🤔

Reply:
See what is acceptable for iOS native: https://github.com/ChartBoost/ios-helium-sdk/pull/1314

Reply:
Yeah, this makes sense, and based on iOS it's how is supposed to be done. it's just massive 😆